cfour/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv, lib, makeWrapper, gfortran, requireFile, writeTextFile, python3, blas-i8, lapack,
```

Comment:
I am surprised that mixing `blas-i8` and `lapack` works here since they are API incompatible (ILP64 vs LP64).

Reply:
The mixture of the i-8 and i-4 interfaces still produce a lot of headache for me. My understanding from what usually happens, is that for BLAS the API often allows to select i-8 or i-4, while LAPACK is not influenced by this. As far as I could see, this it what happens for OpenBLAS, when you switch to the 64 bit integers. At least it was when I did it outside of Nix. I have no idea how deep the LAPACK overrides in NixPkgs go. It is a version that I commonly saw: NWChem with i-8 OpenBLAS can still us i-4 LAPACK, the same for OpenMolcas, where actually ScaLAPACK is involved, which is more or less always i-4. There is a port of ScaLAPACK for i-8, which requires MPI with i-8 (which breaks all packages, for which I've ever tried) or Intel's ScaLAPACK, which seems to be the only ScaLAPACK that supports a working i-8. I tried for a long time with Molcas (see [here](https://gitlab.com/Molcas/OpenMolcas/-/issues/159)). 
In the cases of NWChem and Molcas, also the underlying GlobalArrays allows to mix i-8 BLAS with i-4 LAPACK (when I build against OpenBLAS with INTERFACE=64 for example) and works fine. It does definitely not when using both in i-8 versions.

Reply:
I think both should be the same API. When you at the NWChem derivation it uses `blas-i8` and `lapack-i8`, which works. Whenever Fortran  defines a bare integer it will depend on the compiler flags if it is i4 or i8. In BLAS and LAPACK this is used to define matrix/vector sizes in the API. LAPACK will hand matrices down to BLAS. If both use different integer sizes I would expect it to crash. I would say that they should always be the same, which should be enforced by an `assert`. Everything should be complied with the same convention: application, LAPACK, BLAS.
Btw. the SCALAPACK reference implementation only works with I4. I never managed to compile it with I8. However, Intel provides a version with I8.
Regarding MOLCAS and GA: I have given up for now compiling it with MPI (I guess I ran exactly into the problem that you are referencing).